core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/TemporarySpeedLimitManager.kt

```diff
@@ -0,0 +1,11 @@
+package fr.sncf.osrd.sim_infra.impl
+
+import fr.sncf.osrd.sim_infra.api.DirTrackChunkId
+import fr.sncf.osrd.sim_infra.api.SpeedLimitProperty
+import fr.sncf.osrd.utils.DistanceRangeMap
+
+class TemporarySpeedLimitManager(
+    val speedLimits: MutableMap<DirTrackChunkId, DistanceRangeMap<SpeedLimitProperty>>,
```

Comment:
```suggestion
    val speedLimits: Map<DirTrackChunkId, DistanceRangeMap<SpeedLimitProperty>>,
```

There's no need for mutability once the parsing is done.

This seems to compile just fine with just changing the type here.

Reply:
Fixed in https://github.com/OpenRailAssociation/osrd/pull/9837/commits/356b2a22b3bb73786f7321b8a5ed694e0cc9e65d